### PR TITLE
Add bootstrap timeout

### LINKED ISSUE #13728: Restart node if it is stuck in bootstrap for more than 2 hours

### PR DESCRIPTION
**Pull request by psteckler**
Add a bootstrap timeout to the `Incremental` observer. After 120 minutes, crash the daemon.

Tested by
- temporarily inverting the `is_seed` condition in `create_sync_status_observer`
- temporarily setting the timeout to less than a second
- running the daemon as a seed
- temporarily not stopping the bootstrap timeout when the status changes to `Synced` (which is almost instantaneous in the local seed node)

Saw: 
```
[FATAL] *** Mina daemon has been stuck in bootstrap for too long ***
*** Shutting down ***
```

Closes #13728.

### CODE REVIEW

**Comment by psteckler** [comment]
!c--build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/mina_lib/mina_lib.ml [hunk 1]
```diff
@@ -556,16 +562,46 @@ let create_sync_status_observer ~logger ~is_seed ~demo_mode ~net
...
+    in
+    let stop_bootstrap_timeout () =
+      match !bootstrap_timeout with
+      | Some timeout ->
+          Timeout.cancel () timeout () ;
+          bootstrap_timeout := None
+      | None ->
+          ()
+    in
+    let handle_status_change state status =
```
Comment (deepthiskumar) [inline src/lib/mina_lib/mina_lib.ml:588]: Doesn't `status` here include `Bootstrap` based on how `incremental_status` is defined [here](https://github.com/MinaProtocol/mina/blob/45e909764f645f85cc88cb9a465367b3e1a5a10f/src/lib/mina_lib/mina_lib.ml#L462) ?  How is `state` tracked?

Comment (psteckler) [reply src/lib/mina_lib/mina_lib.ml:588]: Yes, I got fooled by the type that merlin was reporting. We only need `status`.

**Comment by psteckler** [comment]
!ci-build-me

**Review by dkijania — APPROVED** [review]

**Review by deepthiskumar — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**